bindings/c/test/apitester/fdb_c_api_tester.cpp

```diff
@@ -348,7 +348,7 @@ bool runWorkloads(TesterOptions& options) {
 		txExecOptions.databasePerTransaction = options.testSpec.databasePerTransaction;
 		txExecOptions.transactionRetryLimit = options.transactionRetryLimit;
 
-		std::unique_ptr<IScheduler> scheduler = createScheduler(options.numClientThreads);
+		std::unique_ptr<IScheduler> scheduler = createScheduler(2 * options.numClientThreads);
```

Comment:
You probably wanted to have here numClients * 2. In any case I think it is better to make necessary adjustments in randomizeOptions. Note, that sometimes we may want to run tests with a single client thread for checking if a certain error is caused due to concurrency. You probably now also do not need it at all after removing the blocking waits.

Reply:
I still do blocking wait for test cases when `blockOnFutures=true`. You're that even my proposed change is not safe, because the number of threads I need depends on the number of possibly concurrently blocking futures which block until something else is scheduled on the same threadpool (set to newVal). I'm not sure what's the best way to handle this. 2*numClients is safe only when every client does at most 1 blocking watch at a time. 

By adjustments in randomizeOptions you meant to add a new option "numSchedulerThread"?

Reply:
I would suggest to introduce another option, something like containsSelfBlockingWorkloads and then do something like:

if (containsBlockingWorkloads && numClientThreads <= numClients) {
    numClientThreads = numClients + 1
    print a warning that numClientThreads was adjusted
 }
 
Alternatively containsBlockingWorkloads could be determined automatically from the list of the workloads